query/raw_result.go

```diff
@@ -7,109 +7,52 @@ import (
 )
 
 // RawResult is a single raw result returned from a raw query.
-// TODO(xichen): Implement `MarshalJSON` to only marshal the `Data` field without the `data tag.
+// TODO(xichen): Implement `MarshalJSON` to only marshal the `Data` field without the `data` tag.
 type RawResult struct {
-	Data string
-
-	// Fields for joining and sorting purposes. These fields are empty for unsorted raw results.
-	DocID         int32
-	OrderByValues []field.ValueUnion
-
-	// This is the index of the raw result when the raw results are ordered as dictated by the
-	// query (e.g., if there are two raw results sorted by time in descending order, then the
-	// first raw result has an `OrderIdx` of 0, and the second one has an `OrderIdx` of 1).
-	OrderIdx int
-	HasData  bool
-}
-
-// RawResultsByDocIDAsc sorts a list of raw results by their doc IDs in ascending order.
-type RawResultsByDocIDAsc []RawResult
-
-func (a RawResultsByDocIDAsc) Len() int           { return len(a) }
-func (a RawResultsByDocIDAsc) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
-func (a RawResultsByDocIDAsc) Less(i, j int) bool { return a[i].DocID < a[j].DocID }
-
-// RawResultsByOrderIdxAsc sorts a list of doc ID values by their order indices in ascending order.
-// NB(xichen): If an item does not have `HasData` set, it'll be at the end of the array after sorting.
-// The ordering between two items neither of which has `HasData` set is non deterministic.
-type RawResultsByOrderIdxAsc []RawResult
-
-func (a RawResultsByOrderIdxAsc) Len() int      { return len(a) }
-func (a RawResultsByOrderIdxAsc) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
-
-func (a RawResultsByOrderIdxAsc) Less(i, j int) bool {
-	if !a[i].HasData {
-		return false
-	}
-	if !a[j].HasData {
-		return true
-	}
-	return a[i].OrderIdx < a[j].OrderIdx
-}
-
-// RawResultIterator is a raw result iterator.
-type RawResultIterator struct {
-	resultsByDocIDAsc []RawResult
-
-	currIdx int
-}
-
-// NewRawResultIterator creates a new raw result iterator.
-func NewRawResultIterator(resultsByDocIDAsc []RawResult) *RawResultIterator {
-	return &RawResultIterator{
-		resultsByDocIDAsc: resultsByDocIDAsc,
-		currIdx:           -1,
-	}
-}
-
-// Next returns true if there are more results to be iterated over.
-func (it *RawResultIterator) Next() bool {
-	if it.currIdx >= len(it.resultsByDocIDAsc) {
-		return false
-	}
-	it.currIdx++
-	return it.currIdx < len(it.resultsByDocIDAsc)
-}
-
-// DocID returns the doc ID of the current raw result.
-func (it *RawResultIterator) DocID() int32 { return it.resultsByDocIDAsc[it.currIdx].DocID }
-
-// Values returns the collection of values to order the current raw results by.
-func (it *RawResultIterator) Values() []field.ValueUnion {
-	return it.resultsByDocIDAsc[it.currIdx].OrderByValues
+	Data          string
```

Comment:
what is data here? the raw source field? If so, can we rename it or leave a comment?

Reply:
Yeah can leave a comment.